src/api/firebase.js

```diff
@@ -131,3 +131,36 @@ export async function checkItem(listId) {
 	const existingList = await getDocs(listCollectionRef);
 	return existingList.empty ? false : true;
 }
+
+export async function comparePurchaseUrgency(data) {
+	const today = new Date();
+	const inactiveList = data.filter(
+		(eachItem) =>
+			eachItem.dateLastPurchased !== null &&
+			numOfDaysBtwnDates(eachItem.dateLastPurchased.toDate(), today) >= 60,
+	);
+
+	sortByUrgencyAndName(inactiveList);
+
+	const activeList = data.filter(
+		(eachItem) =>
+			eachItem.dateLastPurchased === null ||
+			numOfDaysBtwnDates(eachItem.dateLastPurchased.toDate(), today) < 60,
+	);
+
+	sortByUrgencyAndName(activeList);
+
+	return activeList.concat(inactiveList);
+}
+
+const sortByUrgencyAndName = (list) => {
```

Comment:
I think that this `sortByUrgencyAndName` function is giving us the bug? If we comment this out, we don't get the bug...

I noticed that this issue is stemming from `prevDateLastPurchased` being updated incorrectly. Sometime `prevDateLastPurchased` is incorrectly update to `null` and that leads to a bunch of issues because that value should alway exist as it gets calculated when the item is added.

I wonder if we're messing up the data in the sorting process... 🤔